app/components/goal-block/component.js

```diff
@@ -40,17 +44,23 @@ export default Component.extend(SwipableListItemMixin, {
     this.set('isActive', false);
   },
 
-  observePanOpen: observer('isPanOpen', function() {
-    if (this.get('isPanOpen')) {
+  setOptionWidth: observer('isEditing', function() {
+    if (this.get('isEditing')) {
       this.setProperties({
-        isEditing: false,
-        isEditingDisabled: true
+        optionWidth: SingleOptionWidth,
+        hasOneOption: true,
+        hasThreeOptions: false
       });
     } else {
       this.setProperties({
-        isEditingDisabled: false
+        optionWidth: SingleOptionWidth * 3,
+        hasOneOption: false,
+        hasThreeOptions: true
       });
     }
-  })
+  }),
 
+  observePanOpen: observer('isPanOpen', function() {
+    return this.get('isPanOpen') ? this.set('isEditingDisabled', true) : this.set('isEditingDisabled', false);
+  })
```

Comment:
If you're going for short-hand, you could just set isEditingDisabled to the negative of isPanOpen. Also, no need to return anything.